### ISSUE TITLE
Minimal server implementation

### ISSUE THREAD

**Issue by jdidion** [jeromekelleher/htsget#38]
It would be very useful to have a minimal server implementation for testing (both unit and integration). I propose:

* Minimal implementation that satisfies all MUST conditions in the spec
* Implements new features suggested in v1.1 of the spec, including URL classes
* Runnable with zero configuration (serve files from cwd on default port)
* Runnable both programmatically and via simple CLI

I will begin working on this. Please let me know if you will accept a PR or if I should plan to host this in a separate repository.

**Comment by jeromekelleher** [comment]
I think this is a great idea @jdidion, and I think it would be a real boost to the community to have such a useful resource. However, I think it would be better to keep the server in its own repo, for a number of reasons:

1. For client users of the current software, adding a server would add a lot of extra dependencies that they don't need.
2. It would add a lot of extra complexity to the current repo.
3. As the client and server would likely not share any actual code, it doesn't seem like there would be anything to be gained from keeping them in the same repo, other than complicating the layout for both.

So basically, I'm very enthusiastic about the idea, but I think it would be better off hosted in its own repo. I'm very happy to link to the server repo etc from the README and so on.

**Comment by jdidion** [comment]
Great! Thanks for considering. I'll let you know when a working version is available to link.

**Comment by jeromekelleher** [comment]
OK, thanks. Closing this issue then.
